library/src/main/java/com/vinted/coper/CoperImpl.kt

```diff
@@ -59,6 +70,53 @@ internal class CoperImpl(private val fragmentManager: FragmentManager) : Coper {
         }
     }
 
+    @Synchronized
```

Comment:
`@Synchronized` protects resources to be accessed by two threads simulationsly. However, suspended coroutine exists protected scope and allows another thread to work with it. Very likely you need to protect it with mutex